app/pdfs/ticket_pdf.rb

```diff
@@ -77,5 +77,9 @@ def draw_third_square
     move_up 180
   end
 
-  def draw_fourth_square; end
+  def draw_fourth_square
+    x1 = @right - @mid_horizontal - 180
+    y1 = bounds.top - @mid_vertical - 180
+    print_qr_code(@physical_ticket.token, pos: [@mid_horizontal + x1 / 2, cursor - y1 / 2], extent: 180, stroke: false)
```

Comment:
If you calculate the x and the y before for clarifying the code... why not something like this?

```ruby
x = @mid_horizontal + (@right - @mid_horizontal - 180) / 2
y = cursor - (bounds.top - @mid_vertical - 180) / 2
print_qr_code(@physical_ticket.token, pos: [x, y], extent: 180, stroke: false)
```